velox/exec/tests/SpillTest.cpp

```diff
@@ -679,3 +680,69 @@ INSTANTIATE_TEST_SUITE_P(
         common::CompressionKind::CompressionKind_ZSTD,
         common::CompressionKind::CompressionKind_LZ4,
         common::CompressionKind::CompressionKind_GZIP));
+
+TEST(SpillTest, spillStats) {
+  SpillStats stats1;
+  stats1.spillRuns = 100;
+  stats1.spilledBytes = 1024;
+  stats1.spilledPartitions = 1024;
+  stats1.spilledFiles = 1023;
+  stats1.spillWriteTimeUs = 1023;
+  stats1.spillFlushTimeUs = 1023;
+  stats1.spillDiskWrites = 1023;
+  stats1.spillSortTimeUs = 1023;
+  stats1.spillFillTimeUs = 1023;
+  stats1.spilledRows = 1023;
+  stats1.spillSerializationTimeUs = 1023;
+  SpillStats stats2;
+  stats2.spillRuns = 100;
+  stats2.spilledBytes = 1024;
+  stats2.spilledPartitions = 1025;
+  stats2.spilledFiles = 1026;
+  stats2.spillWriteTimeUs = 1026;
+  stats2.spillFlushTimeUs = 1027;
+  stats2.spillDiskWrites = 1028;
+  stats2.spillSortTimeUs = 1029;
+  stats2.spillFillTimeUs = 1030;
+  stats2.spilledRows = 1031;
+  stats2.spillSerializationTimeUs = 1032;
+  ASSERT_LT(stats1, stats2);
+  ASSERT_GT(stats2, stats1);
+  ASSERT_GE(stats1, stats1);
+  ASSERT_LE(stats1, stats1);
+  ASSERT_NE(stats1, stats2);
+  SpillStats delta = stats2 - stats1;
+  ASSERT_EQ(delta.spilledBytes, 0);
+  ASSERT_EQ(delta.spilledPartitions, 1);
+  ASSERT_EQ(delta.spilledFiles, 3);
+  ASSERT_EQ(delta.spillWriteTimeUs, 3);
+  ASSERT_EQ(delta.spillFlushTimeUs, 4);
+  ASSERT_EQ(delta.spillDiskWrites, 5);
+  ASSERT_EQ(delta.spillSortTimeUs, 6);
+  ASSERT_EQ(delta.spillFillTimeUs, 7);
+  ASSERT_EQ(delta.spilledRows, 8);
+  ASSERT_EQ(delta.spillSerializationTimeUs, 9);
+  delta = stats1 - stats2;
+  ASSERT_EQ(delta.spilledBytes, 0);
+  ASSERT_EQ(delta.spilledPartitions, -1);
+  ASSERT_EQ(delta.spilledFiles, -3);
+  ASSERT_EQ(delta.spillWriteTimeUs, -3);
+  ASSERT_EQ(delta.spillFlushTimeUs, -4);
+  ASSERT_EQ(delta.spillDiskWrites, -5);
+  ASSERT_EQ(delta.spillSortTimeUs, -6);
+  ASSERT_EQ(delta.spillFillTimeUs, -7);
+  ASSERT_EQ(delta.spilledRows, -8);
+  ASSERT_EQ(delta.spillSerializationTimeUs, -9);
+  stats1.spilledBytes = 1030;
+  VELOX_ASSERT_THROW(stats1 < stats2, "");
+  VELOX_ASSERT_THROW(stats1 > stats2, "");
+  VELOX_ASSERT_THROW(stats1 <= stats2, "");
+  VELOX_ASSERT_THROW(stats1 >= stats2, "");
+  ASSERT_NE(stats1, stats2);
+  const SpillStats zeroStats;
+  stats1.reset();
+  ASSERT_EQ(zeroStats, stats1);
+  ASSERT_EQ(
+      stats2.toString(),
+      "spillRuns[100] spilledBytes[1.00KB] spilledRows[1031] spilledPartitions[1025] spilledFiles[1026] spillFillTimeUs[1.03ms] spillSortTime[1.03ms] spillSerializationTime[1.03ms] spillDiskWrites[1028] spillFlushTime[1.03ms] spillWriteTime[1.03ms]");
+}
```

Comment:
Do you think we can add some logic check? like:
a = ...
b = ...
assert_true(a < b)
assert_true(a <= b)
assert_false(a >= b)
assert_false(a > b)
assert_false(a == b)
assert_true(a != b)

Reply:
Good point!